projects/hello_buttons_adcs/src/main.c

```diff
@@ -0,0 +1,59 @@
+#include <stdint.h>
+
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+
+static GpioAddress adc_pin_addr = { .port = GPIO_PORT_A, .pin = 6 };
+static GpioAddress button_addr = { .port = GPIO_PORT_B, .pin = 2 };
+
+static GpioSettings adc_pin_settings = {
+  GPIO_DIR_IN,
+  GPIO_STATE_LOW,
+  GPIO_RES_NONE,
+  GPIO_ALTFN_ANALOG,
```

Comment:
Please use the GNU-style `.direction = GPIO_DIR_IN`, `.state = GPIO_STATE_LOW`, etc for initializing settings structs like this. It makes it easier to read and lets you move around the lines without affecting the meaning.